### PR TITLE
Helper hashmaps for black hole gen

### PR DESCRIPTION
**Pull request by Connor-Colenso**
For use in eye of harmony and potentially cleaning up VM code / other uses later on.

### CODE REVIEW

File: src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java [hunk 1]
```diff
@@ -26,6 +45,40 @@ public GT5OreLayerHelper() {
...
+                dimToOreWrapper.get(dim).calculateWeights();
+            }
+
+            for (String dim : dimToOreWrapper.keySet()) {
+                double a = dimToOreWrapper.get(dim).oreVeinToProbabilityInDimension.values()
+                .stream()
+                .mapToDouble(Double::valueOf)
+                .sum();
+
+                System.out.println("TEST312IMJD + " + a);
```
Comment (glowredman) [inline src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java]: I guess you forgot to remove this...

Comment (Connor-Colenso) [reply src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java]: Hehe

File: src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java [hunk 2]
```diff
@@ -12,11 +14,28 @@
 
 public class GT5OreLayerHelper {
 
+    class oreDimensionWrapper {
```
Comment (glowredman) [inline src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java]: 1) Per convention, Java class names always start with an uppercase letter.
2) Please declare the class as private, if you don't intend it to be used outside of `GT5OreLayerHelper`.

File: src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java [hunk 3]
```diff
@@ -12,11 +14,28 @@
...
+        ArrayList<OreLayerWrapper> internalDimOreList = new ArrayList<>();
+        HashMap<OreLayerWrapper, Double> oreVeinToProbabilityInDimension = new HashMap<OreLayerWrapper, Double>();
```
Comment (glowredman) [inline src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java]: 1.
 ```java
List<OreLayerWrapper> internalDimOreList = new ArrayList<>();
Map<OreLayerWrapper, Double> oreVeinToProbabilityInDimension = new HashMap<>();
```
2. this, too, should be either public or private
3. and probably final too

File: src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java [hunk 4]
```diff
@@ -12,11 +14,28 @@
...
+
+        // Calculate all weights of ore veins once dimension is initialised.
+        void calculateWeights() {
```
Comment (glowredman) [inline src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java]: public or private...

File: src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java [hunk 5]
```diff
@@ -26,6 +45,30 @@ public GT5OreLayerHelper() {
         for (OreLayerWrapper layer : mapOreLayerWrapper.values()) {
             bufferedDims.put(layer, getDims(layer));
         }
+
+        // ------------------------------
+        // Get dims as "Ow,Ne,Ma" etc.
+        for (OreLayerWrapper oreLayer : bufferedDims.keySet()) {
```
Comment (glowredman) [inline src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java]: I'd prefer
```java
for (Map.Entry<OreLayerWrapper, String> entry : bufferedDims.entrySet()) {
    ...
}
```
or
```java
bufferedDims.forEach((oreLayer, dims) -> {
    ...
});
```

File: src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java [hunk 6]
```diff
@@ -39,12 +82,40 @@ public static class OreLayerWrapper {
...
             this.Meta[1] = worldGen.mSecondaryMeta;
             this.Meta[2] = worldGen.mBetweenMeta;
             this.Meta[3] = worldGen.mSporadicMeta;
+
+            // Black magic, don't ask me how it works, I have no idea.
+            try {
+                this.mPrimaryVeinMaterial = GT_OreDictUnificator.getAssociation(
+                                new ItemStack(GregTech_API.sBlockOres1, 1, worldGen.mPrimaryMeta))
+                        .mMaterial
+                        .mMaterial;
```
Comment (glowredman) [inline src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java:100]: Why `mMaterial` twice?

Comment (Connor-Colenso) [reply src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java:100]: Old spaghetti code presumably by Greg

File: src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java [hunk 7]
```diff
@@ -26,6 +45,30 @@ public GT5OreLayerHelper() {
...
             bufferedDims.put(layer, getDims(layer));
         }
+
+        // ------------------------------
+        // Get dims as "Ow,Ne,Ma" etc.
+        for (OreLayerWrapper oreLayer : bufferedDims.keySet()) {
+
+            String dims = bufferedDims.get(oreLayer);
+
+            if (dims.equals("Not available in any Galactic Dim!")) {
```
Comment (glowredman) [inline src/main/java/pers/gwyog/gtneioreplugin/util/GT5OreLayerHelper.java]: I guess this exact String can be found at least at one other place, so it's a good idea to define a constant here or there, so shit doesn't break if one or the other is changed.

**Review by glowredman — APPROVED** [review]

**Pull request merged**